### PR TITLE
Configuration object + Separate request method

### PR DESCRIPTION
**Pull request by wallin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,12 +11,30 @@ gem 'castle-middleware'
...
-Castle::Middleware.api_secret = 'O2ygKmdFTWSqhAF1SL1UyttyEiHCz9F'
+Castle::Middleware.configure do |config|
+  config.app_id = '186593875646714'
+  config.pub_key = 'pk_123456789abcdf'
+end
 ```
 
 The middleware will insert itself into the Rack middleware stack.
 
+### Manually activate middleware
```
Comment (brissmyr) [inline README.md]: "activate" -> "insert"

File: README.md [hunk 2]
```diff
@@ -11,12 +11,30 @@ gem 'castle-middleware'
...
+
+```ruby
+# In castle initializer
+
+config.auto_inject = false
```
Comment (brissmyr) [inline README.md]: `auto_insert_middleware` to not confuse with something like `auto_inject_js`?

Comment (brissmyr) [reply README.md]: Alternatively, we let people define the middleware to insert after, e.g. `config.insert_after = "ActionDispatch::Flash`. But not important right now. FYI, this option is only applicable to Rails apps.

Comment (wallin) [reply README.md]: Good point. That name makes more sense. 

Yeah, we can figure that out. Just though we should handle the case if someone isn't using the flash middleware. (like in our own API)

**Pull request merged**